src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,447 @@
+        ...  coords={"lat":numpy.array([90, 89, 88, 87]),
+        ...          "lon":numpy.array([70, 69, 68, 67])}
+        ... )
+        >>> da.aat._check_coords_inverted()
+        (True, False)
+        """
+        data_obj = self._get_obj(inplace=False)
+        lat = data_obj.get_index(self.y_dim)
+        lat_start = lat[0]
+        lat_end = lat[-1]
+        lon = data_obj.get_index(self.x_dim)
+        lon_start = lon[0]
+        lon_end = lon[-1]
+        return lon_start > lon_end, lat_start < lat_end
+
+    def change_longitude_range(
+        self, inplace: bool = False
+    ) -> xarray.DataArray:
+        """Convert longitude range between -180 to 180 and 0 to 360.
+
+        For some raster data, outputs are incorrect if longitude ranges
+        are not converted from 0 to 360 to -180 to 180, such as the IRI
+        seasonal forecast, but *not* the IRI CAMS observational terciles,
+        which are incorrect if ranges are stored as -180 to 180.
+
+        ``change_longitude_range()`` will convert between the
+        two coordinate systems based on its current state. If coordinates
+        lie solely between 0 and 180 then there is no need for conversion
+        and the input  will be returned.
+
+        Parameters
+        ----------
+        inplace : bool, optional
+            If True, will overwrite existing data array. Default is False.
+
+        Returns
+        -------
+        xarray.DataArray
+            Dataset with transformed longitude coordinates.
+
+        Examples
+        --------
+        >>> import xarray
+        >>> import numpy
+        >>> da = xarray.DataArray(
+        ...  numpy.arange(16).reshape(4,4),
+        ...  coords={"lat":numpy.array([87, 88, 89, 90]),
+        ...          "lon":numpy.array([5, 120, 199, 360])}
+        ... )
+        >>> da_inv = da.aat.change_longitude_range()
+        >>> da_inv.get_index("lon")
+        Int64Index([-161, 0, 5, 120], dtype='int64', name='lon')
+        """
+        data_obj = self._get_obj(inplace=inplace)
+        lon_min = data_obj.indexes[self.x_dim].min()
+        lon_max = data_obj.indexes[self.x_dim].max()
+
+        if lon_max > 180:
+            logger.info("Converting longitude from 0 360 to -180 to 180.")
+
+            data_obj = data_obj.assign_coords(
+                {self.x_dim: ((data_obj[self.x_dim] + 180) % 360) - 180}
+            ).sortby(self.x_dim)
+
+        elif lon_min < 0:
+            logger.info("Converting longitude from -180 to 180 to 0 to 360.")
+
+            data_obj = data_obj.assign_coords(
+                {
+                    self.x_dim: np.where(  # noqa: FKA01
+                        data_obj[self.x_dim] < 0,
+                        data_obj[self.x_dim] + 360,
+                        data_obj[self.x_dim],
+                    )
+                }
+            ).sortby(self.x_dim)
+
+        else:
+            logger.info(
+                "Indeterminate longitude range and no need to convert."
```

Comment:
I think this is not fully correct though.. Cause if all values are between 0 and 180, you don't know if the full range the coordinates are in -180 to 180 or 0 to 360. So maybe it is now in -180 to 180 and to shift it to 0-360 we would have to add 180.. 
It could be solved by giving as input arg the current coordinate range? But maybe also overcomplicating

Reply:
@Tinkaa Conversions between -180 to 180 and 0 to 360 only convert values that are `< 0` in the first instance or `> 180` in the second. Values between `0` and `180` are identical in both coordinate systems, hence why no shift would need to happen. I had to look this up and confirm myself while thinking through the implemented method ha. [This website](http://gis.humboldt.edu/OLM/Courses/old/GSP_101/01_IntroAndReivew/Presentation/Latitude_and_Longitude.html) helped me wrap my around it.